src/plugins/dsn/plugin.js

```diff
@@ -0,0 +1,374 @@
+                        : power;
+                },
+                validate: function (power) {
+                    return !isNaN(parseFloat(power));
+                }
+            });
+
+            // Target formatters
+            openmct.telemetry.addFormat({
+                key: 'range-to-string',
+                format: function (distance) {
+                    // Assumes distance is in km
+                    if (typeof distance === 'number') {
+                        if (distance >= 0) {
+                            if (distance < 1000) {
+                                return distance.toFixed(2) + ' km';
+                            } else if (distance < 1000000) {
+                                return (distance / 1000).toFixed(2) + ' thousand km';
+                            } else if (distance < 1000000000) {
+                                return (distance / 1000000).toFixed(2) + ' million km';
+                            } else if (distance < 1000000000000) {
+                                return (distance / 1000000000).toFixed(2) + ' billion km';
+                            } else if (distance < 1000000000000000) {
+                                return (distance / 1000000000000).toFixed(2) + ' trillion km';
+                            } else {
+                                return (distance / 1000000000000000).toFixed(2) + ' quadrillion km';
+                            }
+                        } else {
+                            return distance;
+                        }
+                    } else {
+                        return distance;
+                    }
+                },
+                parse: function (distance) {
+                    return typeof distance === 'string'
+                        ? parseFloat(distance)
+                        : distance;
+                },
+                validate: function (distance) {
+                    return !isNaN(parseFloat(distance));
+                }
+            });
+
+            openmct.telemetry.addFormat({
+                key: 'light-time-to-string',
+                format: function (lightTime) {
+                    // Assumes light time is in seconds
+                    if (typeof lightTime === 'number') {
+                        if (lightTime >= 0) {
+                            if (lightTime < 60) {
+                                return lightTime.toFixed(2) + ' sec';
+                            } else if (lightTime < 3600) {
+                                return (lightTime / 60).toFixed(2) + ' minutes';
+                            } else if (lightTime < 86400) {
+                                return (lightTime / 3600).toFixed(2) + ' hours';
+                            } else if (lightTime < 604800) {
+                                return (lightTime / 86400).toFixed(2) + ' days';
+                            } else {
+                                return (lightTime / 604800).toFixed(2) + ' weeks';
```

Comment:
Maybe extract all these ints to `CONST SOME_VAL`

Reply:
@evenstensberg Don't think we're doing ES6 with all of the `var` going on...

Reply:
sorry, so what I'm meaning is: `var CONST_GOD_SPEED = 123`